statuslistener/statuslistener.go

```diff
@@ -86,9 +86,9 @@ func (avs *AvailabilityStatusListener) headersFrom(message kafka.Message) []kafk
 }
 
 func (avs *AvailabilityStatusListener) processEvent(statusMessage types.StatusMessage, headers []kafka.Header) {
-	resourceID, err := util.InterfaceToInt64(statusMessage.ResourceID)
-	if err != nil {
-		logging.Log.Errorf("Error parsing resource_id: %s", err.Error())
+	resource := util.Resource{}
+	if resource.Parse(statusMessage) != nil {
```

Comment:
So for this it's a bit more idiomatic go to have a function like `util.ParseStatusMessageToResource` that returns the resource + an error. I could see the error message being what was logged here.

EDIT: It looks like there is an error being returned already! So we can just change the signature of that function.

Reply:
done 👍